### PR TITLE
Document transport-encryption Eventing feature

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes https://github.com/knative/docs/issues/5525

## Proposed Changes 

- Document transport-encryption feature

### CODE REVIEW

**Comment by ReToCode** [comment]
Do you want to add a navbar item for it as well?

**Comment by matzew** [comment]
I miss an entry here: https://deploy-preview-5642--knative.netlify.app/docs/eventing/experimental-features/new-trigger-filters/

File: docs/eventing/experimental-features/transport-encryption.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+the following:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-features
+  namespace: knative-eventing
+data:
+  transport-encryption: "strict"
```
Comment (matzew) [inline docs/eventing/experimental-features/transport-encryption.md:56]: I did just the `kubectl apply` from here: https://cert-manager.io/docs/installation/

```
k get pods -A
NAMESPACE            NAME                                         READY   STATUS    RESTARTS   AGE
cert-manager         cert-manager-5f68c9c6dd-tcms7                1/1     Running   0          2m8s
cert-manager         cert-manager-cainjector-57d6fc9f7d-kh5jc     1/1     Running   0          2m8s
cert-manager         cert-manager-webhook-5b7ffbdc98-zjzvz        1/1     Running   0          2m8s
```

set the value to `strict`, and created a broker. Result:

```
default     my-broker         1s    False   NoAddress
```

Comment (pierDipi) [reply docs/eventing/experimental-features/transport-encryption.md:56]: I missed that we need to apply the new Eventing TLS artifact from the release assets

Comment (matzew) [reply docs/eventing/experimental-features/transport-encryption.md:56]: I had it installed via `ko apply -f config/tls/issuers`

same (using the upstream 1.11 branch)

Comment (pierDipi) [reply docs/eventing/experimental-features/transport-encryption.md:56]: That's not enough, you need certificates as well

Comment (matzew) [reply docs/eventing/experimental-features/transport-encryption.md:56]: Looks like should be listed in the doc

Comment (matzew) [reply docs/eventing/experimental-features/transport-encryption.md:56]: FWIW, doing: 

```
➜  eventing git:(1a5f37cdf) k apply -f https://github.com/knative/eventing/releases/download/knative-v1.11.0/eventing-tls-networking.yaml
```
fixed it :wink:

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**